src/adapters/okx-avalanche-2/index.ts

```diff
@@ -0,0 +1,111 @@
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+import BigNumber from "bignumber.js";
+import moment from "moment";
+import { EventData } from "web3-eth-contract";
+import path from "path";
+import Avalanche from "../../sdk/avalanche";
+import { ISaleEntity } from "../../sdk/Interfaces";
+import { handleExtraData } from "../okx-polygon-2";
+
+class OKX {
+    name: string;
+    protocol: string;
+    block: number;
+    deprecatedAtBlock: number;
+    contract: string;
+    events: string[];
+    pathToAbi: string;
+    range: number;
+    chunkSize: number;
+    sdk: any;
+
+    constructor() {
+        this.name = "okx-avalanche-1";
```

Comment:
Provider name must match with the folder name (e.g. this should be changed to `okx-avalanche-2`). Please do it with other providers that you have added.